eStore/Core/UI/Controls/SectionedMenuView.swift

```diff
@@ -0,0 +1,117 @@
+//
+//  SectionedMenuView.swift
+//  eStore
+//
+//  Created by Vladislav Kondrashkov on 4/17/19.
+//  Copyright © 2019 Vladislav Kondrashkov. All rights reserved.
+//
+
+import UIKit
+
+enum SectionedMenuRowType {
+    case thumbnail
+    case regular
+    case colorPicker
+}
+
+struct SectionedMenuSection {
+    let title: String?
+    let items: [SectionedMenuRow]
+}
+
+struct SectionedMenuRow {
+    let imageUrl: String?
+    let title: String?
+    let type: SectionedMenuRowType
+    let action: ((_ point: CGPoint) -> Void)?
+
+    init(imageUrl: String?,
+         title: String?,
+         type: SectionedMenuRowType = .regular,
+         action: ((_ point: CGPoint) -> Void)?) {
+
+        self.imageUrl = imageUrl
+        self.title = title
+        self.type = type
+        self.action = action
+    }
+}
+
+final class SectionedMenuView: UITableView {
+    var theme: Theme!
+    var sections: [SectionedMenuSection] = []
+
+    override init(frame: CGRect, style: UITableView.Style) {
+        super.init(frame: frame, style: style)
+
+        register(SectionedMenuRegularCell.self, forCellReuseIdentifier: SectionedMenuRegularCell.reuseIdentifier)
+        register(SectionedMenuColorPickerCell.self, forCellReuseIdentifier: SectionedMenuColorPickerCell.reuseIdentifier)
+        register(SectionedMenuThumbnailCell.self, forCellReuseIdentifier: SectionedMenuThumbnailCell.reuseIdentifier)
+        
+        delegate = self
+        dataSource = self
+
+
+        let sectionedMenuGestureRecognizer = UITapGestureRecognizer(target: self, action: #selector(sectionedMenuDidTap(sender:)))
+        sectionedMenuGestureRecognizer.cancelsTouchesInView = false
+        addGestureRecognizer(sectionedMenuGestureRecognizer)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    @objc func sectionedMenuDidTap(sender: UITapGestureRecognizer) {
+        let location = sender.location(in: self) // point of touch in tableView
+        guard let indexPath = indexPathForRow(at: location) else {
+            return
+        }
+        let windowPoint = self.convert(location, to: self.window)
+        sections[indexPath.section].items[indexPath.row].action?(windowPoint)
+    }
+}
+
+// MARK: - UITableViewDelegate implementation
+// TODO: Make proxying if needed
```

Comment:
Hey, there is no need in proxy at all. It is okay to just move a delegate and datasource to separate object. Though it is usually better to have them in one place... this thing would be something like an adapter. I would say `SectionMenuTableViewAdapter`...